syntax = "proto3";

package inference;

service InferenceService {
  rpc Predict (ImageRequest) returns (PredictionResponse);
  rpc HealthCheck (HealthRequest) returns (HealthResponse);
}

message ImageRequest {
  bytes image_data = 1;  // Send JPEG-encoded image
}

message PredictionResponse {
  string label = 1;
  float confidence = 2;
}

message HealthRequest {}
message HealthResponse {
  string status = 1;
}